game-core/src/test/java/games/strategy/engine/lobby/client/ui/LobbyGamePanelTest.java

```diff
@@ -0,0 +1,85 @@
+package games.strategy.engine.lobby.client.ui;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Spy;
+
+import com.example.mockito.MockitoExtension;
+
+import games.strategy.engine.lobby.server.GameDescription;
+
+public final class LobbyGamePanelTest {
+  @ExtendWith(MockitoExtension.class)
+  @Nested
+  public final class GetBotSupportEmailTest {
+    @Spy
+    private final GameDescription gameDescription = new GameDescription();
+
+    @Test
+    public void shouldReturnBotSupportEmailWhenBotSupportEmailIsNotEmpty() {
+      final String botSupportEmail = "bot@me.com";
+      when(gameDescription.getBotSupportEmail()).thenReturn(botSupportEmail);
+
+      assertThat(LobbyGamePanel.getBotSupportEmail(gameDescription), is(Optional.of(botSupportEmail)));
```

Comment:
Hmm, thinking about the object abstraction, this does not really make sense: `LobbyGamePanel.getBotSupportEmail(gameDescription)`  Why would lobby game panel have a parsing method, to extract data from a `gameDescription`? It would make sense to consume that data, but extraction does not make sense to me. I suspect a feature envy code smell, moving the more model like methods to  gameDescription, or a new view model class should fix. That would remove the non rendering logic from the panel class.